cmd/spire-server/cli/agent/list.go

```diff
@@ -151,3 +160,8 @@ func parseToSelectorMatch(match string) (types.SelectorMatch_MatchBehavior, erro
 		return types.SelectorMatch_MATCH_SUPERSET, errors.New("unsupported match behavior")
 	}
 }
+
+func exitWithError(env *commoncli.Env, err error) {
```

Comment:
Not sure if this is the best approach for maintaining return code behavior here, this will make it difficult  for testing, since the `Run` method is expected to **return** the exit code, and not **evaluate** it. I was thinking in maybe extending the Printer interface to include printing methods that returns the given error, something like:
```
type Printer interface {
	MustPrintError(error)
	MustPrintProto(...proto.Message)
	MustPrintStruct(...interface{})
	PrintError(error) error
	PrintProto(...proto.Message) error
	PrintStruct(...interface{}) error
}
```
Or even to update existing methods to remove os exit logic eval: 
```
type Printer interface {
	PrintError(error) error
	PrintProto(...proto.Message) error
	PrintStruct(...interface{}) error
}
```
What do you think @azdagron?

Reply:
I think the important thing here is consistency. Either we return errors from things that print (that we then check) or we exit(2) internally. Right now we have a mixture. The scope of the original PR that introduced the cliprinter wasn't quite big enough to tackle all of these kinds of details, so some more changes might be necessary. Let me stare at the CLI code for a bit and see what I can see :)

Reply:
As discussed in the contributor sync meeting, I'm updating the printer interface to remove the error-handling wrapper, letting the consumer handle eventual errors with built-in and custom print functions. This will facilitate the migration of our commands since printer errors are already handled by our command adapter (cmd/spire-server/util/util.go:141), which has a return code of 1 instead of 2 in these scenarios.